contracts/distribution/subdomains/abstractions/AZNSPricing.sol

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.18;
+
+
+// TODO sub: abstract or interface?
+abstract contract AZNSPricing {
+    // TODO sub: do we leave only return of the price
+    //  or do we leave fee values here as well?
+    function getPrice(
+        bytes32 parentHash,
+        string calldata label
+    ) external virtual view returns (uint256);
+
+    function feeEnforced() external pure virtual returns (bool) {
+        return false;
+    }
```

Comment:
yeah, still thinking about the structure of these abstracts. I added this for now since user developed modules can override these to not behave properly, this way we ensure that a pricing with fee will always return true here.

Reply:
still not sure what the problem is exactly.

Reply:
It's just that the price logic is separate from getPrice.

It's possible for this AZNSPricing contract to intend to enforce a fee because feeEnforced() returns true, but for ultimately the fee to not get paid because the middleman contract where the fee is then passed into the payment contract isn't doing that correctly for some reason, which isn't happening but just an example of a problem that can arise from the fee logic ultimately being external to the pricing contract like this.